### PR TITLE
feat(rollappparams): added param for min gas price and respective ante handlers

### PR DESCRIPTION
**Pull request by keruch**
…e handlers

## Description

----

Closes #XXX

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

PR review checkboxes:

I have...

- [ ]  Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ]  Targeted PR against the correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to the GitHub issue with discussion and accepted design
- [ ]  Targets only one GitHub issue
- [ ]  Wrote unit and integration tests
- [ ]  Wrote relevant migration scripts if necessary
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)
- [ ]  Add an issue in the [e2e-tests repo](https://github.com/dymensionxyz/e2e-tests) if necessary

SDK Checklist
- [ ] Import/Export Genesis
- [ ] Registered Invariants
- [ ] Registered Events
- [ ] Updated openapi.yaml
- [ ] No usage of go `map`
- [ ] No usage of `time.Now()`
- [ ] Used fixed point arithmetic and not float arithmetic
- [ ] Avoid panicking in Begin/End block as much as possible
- [ ] No unexpected math Overflow
- [ ] Used `sendCoin` and not `SendCoins`
- [ ] Out-of-block compute is bounded
- [ ] No serialized ID at the end of store keys
- [ ] UInt to byte conversion should use BigEndian

Full security checklist [here](https://www.faulttolerant.xyz/2024-01-16-cosmos-security-1/)

----

For Reviewer:

- [ ]  Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  Confirmed all author checklist items have been addressed

----

After reviewer approval:

- [ ]  In case the PR targets the main branch, PR should not be squash merge in order to keep meaningful git history.
- [ ]  In case the PR targets a release branch, PR must be rebased.

### CODE REVIEW

File: x/rollappparams/types/helpers.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	"slices"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// IntersectMinGasPrices returns the intersection of two MinGasPrices.
+// The intersection is the maximum of the two prices for each denom.
+// Operation is commutative.
+func IntersectMinGasPrices(a, b sdk.DecCoins) sdk.DecCoins {
+	res := make([]sdk.DecCoin, 0, len(a))
```
Comment (mtsitrin) [inline x/rollappparams/types/helpers.go:13]:
```suggestion
    res := make([]sdk.DecCoin, 0, min(len(a), len(b)))
```

**Review by mtsitrin — APPROVED** [review]
Great!
approved

**Comment by mtsitrin** [comment]
@omritoptix don't u think this "intersect" logic is too strict?

I think it might be better to allow `local` set to **override** and add on top the global set

e.g
global min denom- `arax`
node that set `arax` in it's local setting - OK. can pay fee in `arax`
node that set `arax` in it's local setting - OK. can pay fee in `arax`
node that set `adym` in it's local setting - won't work (the setting needs to have `adym` + `arax`)

don;t we prefer the accepted denoms to be global + local?

**Comment by omritoptix** [comment]
> @omritoptix don't u think this "intersect" logic is too strict?

Sharing our slack convo:

> not I don’t think it’s good cause I think it’s missing the point of this feature if you allow it.
> basically you tell every node he can put whatever gas token he wants regardless to the decided gov param which than he can generate shit token on some side chain, pass it and use that as gas and basically spam the network which is excatly what we want to avoid.

**Review by omritoptix — APPROVED** [review]

**Pull request merged**